### ISSUE TITLE
Client cannot allocate files

### ISSUE THREAD

**Issue by cpcnw** [ignacionelson/ProjectSend#98]
If a client belongs to a group and uploads a file - only admin can allocate that file to the group.

Is it possible that either a) clients can allocate files to groups they belong in b) files uploaded by clients in groups are automatically available to the group?

Thanks

**Comment by lenamtl** [comment]
This is not possible at the moment.

**Comment by cpcnw** [comment]
Guess thats a feature request then :)

**Comment by lenamtl** [comment]
In which context you need this?
Do you mean that you want to allow any client from a group or allow a specific client to can do that?

**Comment by cpcnw** [comment]
I just think that anyone who is in the group shoulod be able to upload a file, and the file be seen by everyone in the group. I setup Project Send for a company and the user also expected that behaviour. I'm not sure of the piont of groups otherwise?

**Comment by lenamtl** [comment]
When the admin assign a file to a group this is a way to mass assign a file.
So only 1 file to upload and 1 assign action.
This is saving time and space.

Allowing user to upload a file to a group can be an option attach to a group or maybe to a client role.
Maybe a another option could be set like moderation step by the admin, between the upload and the publication.
This should not be allowed by default.

**Comment by cpcnw** [comment]
When working in a team you add your team members to a group. If each team member is uploading 5-10 files a day that makes a big job for admin toa pprove them and is not feasable. Adding someone to a group should mean they are trusted to upload files. Maybe not by default but it should be an option imho.

**Comment by devyljon** [comment]
Hello,

I saw this post and cpcnw is right. 
My company has users who need to share big files with service provider.
So the solution should be to create an accounts for the user company and another one for the service provider.
These two account should be in the same group, and in this way when one upload a file, and assign it to the group, the other person should see files without the approvement of the admin.

It should be the normal behaviour of the application. If we have to wait that the admin assign files to a groups, users can wait a long time before admin see files.

The solution of my problem is to give the same account for both, but it means that I can't assign different right

This feature will be implemented in the futur version of ProjectSend?

(Sorry for my english)

**Comment by wcousin** [comment]
I second this feature. Also, if the client is assigned to more than one group then they should be able to pick which group the file is accessible to on upload and notify that group.

**Comment by lenamtl** [comment]
I think this should be set as an option not enabled by default.
We all have different requests / needs, so when adding a feature like this the super admin should be able to activate the feature or not.  
Deep analysis should be done first to prevent code structure and logical issues.

To my opinion the whole group structure/logical should be rethink.

**Comment by ignacionelson** [comment]
This issue is being closed as part of a bulk cleanup. ProjectSend has undergone significant changes since this was reported (including r1720 and r1994), and many older issues no longer apply to the current codebase.

If you are still experiencing this problem on the latest version, please open a new issue with updated details and we'll be happy to look into it. Thank you for your contribution!
